cnxarchive/tests/views/test_content.py

```diff
@@ -872,7 +872,6 @@ def test_get_extra_no_allowable_types(self):
                          'application/json')
         output['canPublish'].sort()
         self.assertEqual(output, {
-            u'books': [],
```

Comment:
Shouldn't the key be available on each request so that the schema expectations are met? That is, there is a general structure, even if it has null values. Or maybe you are keying in on this a way to decipher that it is not contextual?

Reply:
That would make sense. Would you prefer to keep the empty arrays or use nulls in this case when the query is not even performed?

Reply:
Your call on empty array or null.

TBH I'm casually thinking about this in terms of the swagger schema, which likes a complete structure to be a valid response. Or at least that's what I've experienced.

Reply:
Asked Ross and he would rather we returned the book itself in both cases (contextual page extras and book extras). I'll work on that.

Reply:
Hm maybe book for page and empty array for book would make more sense actually.